odl/operator/pspace_ops.py

```diff
@@ -390,14 +393,47 @@ def __getitem__(self, index):
         0
         >>> prod_op[1, 1]
         0
+
+        By accessing single indices, a row is extracted as a
+        `ReductionOperator`:
+
+        >>> prod_op[0]
+        ReductionOperator(ZeroOperator(rn(3)), IdentityOperator(rn(3)))
         """
-        row, col = index
-        linear_index = np.flatnonzero((self.ops.row == row) &
-                                      (self.ops.col == col))
-        if linear_index.size == 0:
-            return 0
+        if isinstance(index, tuple):
+            row, col = index
+
+            linear_index = np.flatnonzero((self.ops.row == row) &
+                                          (self.ops.col == col))
+            if linear_index.size == 0:
+                return 0
+            else:
+                return self.ops.data[int(linear_index)]
         else:
-            return self.ops.data[int(linear_index)]
+            ops = [None] * len(self.domain)
+            for op, col, row in zip(self.ops.data, self.ops.col, self.ops.row):
+                if row == index:
```

Comment:
Perhaps better to convert to integer first, otherwise you can get these "comparison is ambitous blabla" messages from Numpy.